pkg/apis/core/v1beta1/helper/cloudprofile_test.go

```diff
@@ -22,57 +24,207 @@ var _ = Describe("CloudProfile Helper", func() {
+						StartTime:      ptr.To(metav1.NewTime(now.Add(-4 * time.Hour))),
+					},
+					{
+						Classification: gardencorev1beta1.ClassificationDeprecated,
+						StartTime:      ptr.To(metav1.NewTime(now.Add(-3 * time.Hour))),
+					},
+					{
+						Classification: gardencorev1beta1.ClassificationExpired,
+						StartTime:      ptr.To(metav1.NewTime(now.Add(-1 * time.Hour))),
+					},
+				},
+			})
+			Expect(classification).To(Equal(gardencorev1beta1.ClassificationExpired))
+		})
+
+		It("first lifecycle start time field is optional", func() {
+			classification := CurrentLifecycleClassification(gardencorev1beta1.ExpirableVersion{
+				Version: "1.28.5",
+				Lifecycle: []gardencorev1beta1.LifecycleStage{
+					{
+						Classification: gardencorev1beta1.ClassificationPreview,
+					},
+					{
+						Classification: gardencorev1beta1.ClassificationSupported,
+						StartTime:      ptr.To(metav1.NewTime(now.Add(3 * time.Hour))),
+					},
+					{
+						Classification: gardencorev1beta1.ClassificationDeprecated,
+						StartTime:      ptr.To(metav1.NewTime(now.Add(4 * time.Hour))),
+					},
+					{
+						Classification: gardencorev1beta1.ClassificationExpired,
+						StartTime:      ptr.To(metav1.NewTime(now.Add(5 * time.Hour))),
+					},
+				},
+			})
+			Expect(classification).To(Equal(gardencorev1beta1.ClassificationPreview))
+		})
+
+		It("determining supported for deprecated classification field", func() {
 			classification := CurrentLifecycleClassification(gardencorev1beta1.ExpirableVersion{
-				Version:        "1.28.0",
 				Classification: ptr.To(gardencorev1beta1.ClassificationSupported),
-				ExpirationDate: ptr.To(metav1.NewTime(time.Now().Add(2 * time.Hour))),
+				Version:        "1.28.0",
 			})
 			Expect(classification).To(Equal(gardencorev1beta1.ClassificationSupported))
 		})
 
-		It("supported version has already expired", func() {
+		It("determining expired for deprecated expiration date field", func() {
 			classification := CurrentLifecycleClassification(gardencorev1beta1.ExpirableVersion{
+				ExpirationDate: ptr.To(metav1.NewTime(now.Add(-1 * time.Hour))),
 				Version:        "1.28.0",
-				Classification: ptr.To(gardencorev1beta1.ClassificationSupported),
-				ExpirationDate: ptr.To(metav1.NewTime(time.Now().Add(-2 * time.Hour))),
 			})
 			Expect(classification).To(Equal(gardencorev1beta1.ClassificationExpired))
 		})
+
+		It("determining preview for deprecated classification and expiration date field", func() {
+			classification := CurrentLifecycleClassification(gardencorev1beta1.ExpirableVersion{
+				Classification: ptr.To(gardencorev1beta1.ClassificationPreview),
+				Version:        "1.28.0",
+				ExpirationDate: ptr.To(metav1.NewTime(now.Add(3 * time.Hour))),
+			})
+			Expect(classification).To(Equal(gardencorev1beta1.ClassificationPreview))
+		})
+
+	})
+
+	// TODO:(RAPSNX) Consider change / improve these tests in future, this was a shourtcut to test the function.
```

Comment:
Is this comment still valid?

Also:

```suggestion
	// TODO:(RAPSNX) Consider change / improve these tests in future, this was a shortcut to test the function.
```